src/compiler/scanner.ts

```diff
@@ -3390,7 +3400,7 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                                 error(Diagnostics.Unicode_property_value_expressions_are_only_available_when_the_Unicode_u_flag_or_the_Unicode_Sets_v_flag_is_set, start, pos - start);
                             }
                         }
-                        else if (unicodeMode) {
+                        else if (!annexB) {

```

Comment:
In Annex B, braces after `p` actually should not be parsed at all, but it does provide helpful errors like https://github.com/microsoft/TypeScript/issues/58275#issuecomment-2068174123

Reply:
Not really outdated

Reply:
The outdated label is just because the comment is on an old revision of the PR and GitHub can't figure out where the comment goes after.

Reply:
Agreed. I could’ve removed the review and re-comment but finally chose not to.